src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13722,17 +13834,26 @@ private new struct ResetPoint
             internal readonly TerminatorState TerminatorState;
             internal readonly bool IsInAsync;
             internal readonly bool IsInQuery;
+            // PROTOTYPE: Why do we need this? We don't need something similar for 'partial'
```

Comment:
we need this because partial can look locally to figure out what it means.  it is not affected by outer contexts.  that's not the case for async/queries/field/value.  field/value shuld be treated very closely to async/await.  An outer context pushes us into a context where those tokens now become keywords.

fortunately field/value is also simpler because we never change the value as we go deeper into the accessor.